packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -33,7 +36,68 @@ declare global {
 /**
  * A nimble styled rich text editor
  */
-export class RichTextEditor extends FoundationElement {
+export class RichTextEditor extends FoundationElement implements ErrorPattern {
+    /**
+     * Whether to disable user from editing and interacting with toolbar buttons
+     *
+     * @public
+     * HTML Attribute: disabled
+     */
+    @attr({ attribute: 'disabled', mode: 'boolean' })
+    public disabled = false;
+
+    /**
+     * Whether to hide the footer of the rich text editor
+     *
+     * @public
+     * HTML Attribute: footer-hidden
+     */
+    @attr({ attribute: 'footer-hidden', mode: 'boolean' })
+    public footerHidden = false;
+
+    /**
+     * Whether to grow the editor vertically to fit the content
+     *
+     * @public
+     * HTML Attribute: fit-to-content
+     */
+    @attr({ attribute: 'fit-to-content', mode: 'boolean' })
+    public fitToContent = false;
+
+    /**
+     * Whether to display the error state.
+     *
+     * @public
+     * HTML Attribute: error-visible
+     */
+    @attr({ attribute: 'error-visible', mode: 'boolean' })
+    public errorVisible = false;
+
+    /**
+     * A message explaining why the value is invalid.
+     *
+     * @public
+     * HTML Attribute: error-text
+     */
+    @attr({ attribute: 'error-text' })
+    public errorText?: string;
+
+    /**
+     * @public
+     * HTML Attribute: placeholder
+     */
+    @attr({ attribute: 'placeholder' })
+    public placeholder = '';
+
+    /**
+     * True if the editor is empty, false otherwise
+     *
+     * @public
+     */
+    public get empty(): boolean {
```

Comment:
In order to create a comment in the comments feature, we will disable the OK button when the editor is either empty or contains only whitespace. This is necessary because the backend service does not permit the creation of a comment with just the whitespace. Therefore, it is essential to prevent users from creating a comment unless there is valid content within the editor. I have also updated the spec with the reasoning.

Reply:
We should be making the API decision based on what makes the most sense in the general sense, not because that's what the comments feature needs. But in this case, since you could probably argue for either behavior, I think it's fine to go with that.